at_secondary/at_persistence_secondary_server/lib/src/keystore/hive_keystore.dart

```diff
@@ -7,23 +9,38 @@ import 'package:at_utils/at_logger.dart';
 import 'package:hive/hive.dart';
 
 class HiveKeystore implements SecondaryKeyStore<String, AtData?, AtMetaData?> {
-  final logger = AtSignLogger('HiveKeystore');
+  final AtSignLogger logger = AtSignLogger('HiveKeystore');
 
   var keyStoreHelper = HiveKeyStoreHelper.getInstance();
   var persistenceManager;
   var _commitLog;
+  final HashMap<String, AtMetaData> _metaDataCache = HashMap();
 
   HiveKeystore();
 
   set commitLog(value) {
     _commitLog = value;
   }
 
+  Future<void> initMetaDataCache() async {
+    if (persistenceManager == null || !persistenceManager.getBox().isOpen) {
+      logger.severe(
+          'persistence manager not initialized. skipping metadata caching');
+      return;
+    }
+    var keys = persistenceManager.getBox().keys;
+    await Future.forEach(
```

Comment:
Add some logs for before and after. Logs helps us to see how the cache init took or if it is in progress.